### ISSUE TITLE
Can't install ndk-sysroot

### ISSUE THREAD

**Issue by BruceMoustache** [termux/termux-packages#5016]
I've tried to install ndk-sysroot but that's what I got
-------
$ apt install ndk-sysroot                                              
Reading package lists... Done
Building dependency tree
Reading state information... Done                                      
The following NEW packages will be installed:
  ndk-sysroot
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.         
12 not fully installed or removed.
Need to get 0 B/1574 kB of archives.                                   
After this operation, 22.3 MB of additional disk space will be used.
(Reading database ... 10087 files and directories currently installed.)
Preparing to unpack .../ndk-sysroot_20-7_arm.deb ...
Unpacking ndk-sysroot (20-7) ...
dpkg: error processing archive /data/data/com.termux/files/usr/var/cach
e/apt/archives/ndk-sysroot_20-7_arm.deb (--unpack):
 unable to install new version of '/data/data/com.termux/files/usr/incl
ude/linux/netfilter_ipv6/ip6t_HL.h': No such file or directory
Errors were encountered while processing:
 /data/data/com.termux/files/usr/var/cache/apt/archives/ndk-sysroot_20-
7_arm.deb
E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an err
or code (1)
$

**Comment by RossComputerGuy** [comment]
Use pkg, not apt

**Comment by xeffyr** [comment]
**Unreproducible**.

![20200308_181724](https://user-images.githubusercontent.com/25881154/76166731-6dcf8b80-6169-11ea-855f-81b8da1f4380.jpg)

**Comment by xeffyr** [comment]
Try
```
apt purge ndk-sysroot
apt install ndk-sysroot
```
However, it is likely that something is broken in your Termux installation.

**Comment by BruceMoustache** [comment]
It don't work

**Comment by BruceMoustache** [comment]
What's your termux version?
I install it from f-droid

**Comment by Grimler91** [comment]
Seems to happen on arm devices only (but I can't reproduce it on my arm device). What's the output of `termux-info` for those of you that have this problem?

**Comment by Grimler91** [comment]
Latest update to ndk-sysroot was a month ago (3f7b9a0d53eab0e76982bc71cc8b7fd8602bf8fd) so it is weird if the errors started now a few days ago

**Comment by xeffyr** [comment]
> but I can't reproduce it on my arm device)

I'm unable too, tested on ARM Android 7.0 and AArch64 Android 8.0.

Maybe there is something related to a specific Termux setup ?

**Comment by BruceMoustache** [comment]
No I delete my old setup. Now it's just vanila termux

**Comment by BruceMoustache** [comment]
$ termux-info
Packages CPU architecture:
arm
Subscribed repositories:
# sources.list
deb https://termux.org/packages/ stable main
# science-repo (sources.list.d/science.list)
deb https://dl.bintray.com/grimler/science-packages-24 science stable
# game-repo (sources.list.d/game.list)
deb https://dl.bintray.com/grimler/game-packages-24 games stable       Updatable packages:
All packages up to date                                                Android version:
10
Kernel build information:                                              Linux localhost 3.4.113-gd97e602 #1 SMP PREEMPT Thu Feb 6 10:56:25 UTC 2020 armv7l Android                                                    Device manufacturer:
motorola
Device model:
Moto G
$

**Comment by BruceMoustache** [comment]
I'm using Arrow OS android 10

**Comment by BruceMoustache** [comment]
$ apt purge ndk-sysroot
Reading package lists... Done
Building dependency tree
Reading state information... Done
Package 'ndk-sysroot' is not installed, so not removed
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 libllvm : Depends: ndk-sysroot but it is not going to be installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
$

how can I resolve this?

**Comment by BruceMoustache** [comment]
In the firt time termux says ndk-sysroot isn't a installable candidate

**Comment by BruceMoustache** [comment]
apt remove ndk-sysroot
Reading package lists... Done
Building dependency tree                                               Reading state information... Done
The following packages were automatically installed and are no longer required:
  binutils libffi
Use 'apt autoremove' to remove them.
The following packages will be REMOVED:                                  clang libllvm ndk-sysroot
0 upgraded, 0 newly installed, 3 to remove and 0 not upgraded.         5 not fully installed or removed.
After this operation, 117 MB disk space will be freed.
Do you want to continue? [Y/n]                                         (Reading database ... 7007 files and directories currently installed.)
Removing clang (9.0.1-1) ...
Removing libllvm (9.0.1-1) ...                                         dpkg: error processing package ndk-sysroot (--remove):
 package is in a very bad inconsistent state; you should
 reinstall it before attempting a removal                              Errors were encountered while processing:
 ndk-sysroot                                                           E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)
$

**Comment by BruceMoustache** [comment]
This APT has Super Cow Powers.
$ apt install --reinstall ndk-sysroot
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages were automatically installed and are no longer required:
  binutils libffi
Use 'apt autoremove' to remove them.
The following packages will be upgraded:
  ndk-sysroot
1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
3 not fully installed or removed.
Need to get 0 B/1574 kB of archives.
After this operation, 22.3 MB of additional disk space will be used.
(Reading database ... 4677 files and directories currently installed.)
Preparing to unpack .../ndk-sysroot_20-7_arm.deb ...
Unpacking ndk-sysroot (20-7) over (20-7) ...
dpkg: error processing archive /data/data/com.termux/files/usr/var/cache/apt/archives/ndk-sysroot_20-7_arm.deb (--unpack):
 unable to install new version of '/data/data/com.termux/files/usr/include/linux/netfilter_ipv6/ip6t_HL.h': No such file or directory
dpkg: error while cleaning up:
 unable to restore backup version of '/data/data/com.termux/files/usr/include/linux/netfilter_ipv6/ip6t_hl.h': No such file or directory
Errors were encountered while processing:
 /data/data/com.termux/files/usr/var/cache/apt/archives/ndk-sysroot_20-7_arm.deb
E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)
$

**Comment by Grimler91** [comment]
@BruceMoustache I have bumped the package. Please see if `apt update && apt install ndk-sysroot` works

**Comment by BruceMoustache** [comment]
[![Screenshot-Termux-20200313-214254.png](https://i.postimg.cc/NG6dpBL4/Screenshot-Termux-20200313-214254.png)](https://postimg.cc/bdv0vXxD)
After clean termux storage I get this message
Even if I reinstall it

**Comment by Grimler91** [comment]
@BruceMoustache are you sure that you are not trying to install termux to the external storage or something? From where are you installing termux? (Google play or fdroid?)

**Comment by BruceMoustache** [comment]
Google play

**Comment by xeffyr** [comment]
Use F-Droid instead as Google Play source provides only split apks. If you got split apk for ARM, it will not work on AArch64.

**Comment by BruceMoustache** [comment]
The same again with F-Droid :-(

**Comment by xeffyr** [comment]
Then only possible reason is problem with file system (out of inodes, corrupted or bug in kernel which is quite common in "tweaked" ones from XDA).

**Comment by EdoZullo** [comment]
Hi,
I have the same issue, can't install ndk-sysroot

$ apt install ndk-sysroot
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  ndk-sysroot
0 upgraded, 1 newly installed, 0 to remove and 17 not upgraded.
Need to get 0 B/1580 kB of archives.
After this operation, 22.3 MB of additional disk space will be used.
(Reading database ... 3936 files and directories currently installed.)
Preparing to unpack .../ndk-sysroot_20-8_arm.deb ...
Unpacking ndk-sysroot (20-8) ...
dpkg: error processing archive /data/data/com.termux/files/usr/var/cache/apt/archives/ndk-sysroot_20-8_arm.deb (--unpack):
 unable to install new version of '/data/data/com.termux/files/usr/include/linux/netfilter/xt_DSCP.h': No such file or directory
Errors were encountered while processing:
 /data/data/com.termux/files/usr/var/cache/apt/archives/ndk-sysroot_20-8_arm.deb
E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)

Packages CPU architecture:
arm
Android version:
9
Kernel build information:
Linux localhost 3.4.113-g263ce0f96fc #1 SMP PREEMPT Sun Apr 21 19:18:10 CEST 2019 armv7l Android
Device manufacturer:
motorola
Device model:
XT1039

**Comment by EdoZullo** [comment]
Please help me.

**Comment by firefresh** [comment]
> Hi,
> I have the same issue, can't install ndk-sysroot. i try to download the pkg and install manually with "dpkg -i" command but the result is the same:
> 
> $ apt install ndk-sysroot
> Reading package lists... Done
> Building dependency tree
> Reading state information... Done
> The following NEW packages will be installed:
> ndk-sysroot
> 0 upgraded, 1 newly installed, 0 to remove and 17 not upgraded.
> Need to get 0 B/1580 kB of archives.
> After this operation, 22.3 MB of additional disk space will be used.
> (Reading database ... 3936 files and directories currently installed.)
> Preparing to unpack .../ndk-sysroot_20-8_arm.deb ...
> Unpacking ndk-sysroot (20-8) ...
> dpkg: error processing archive /data/data/com.termux/files/usr/var/cache/apt/archives/ndk-sysroot_20-8_arm.deb (--unpack):
> unable to install new version of '/data/data/com.termux/files/usr/include/linux/netfilter/xt_DSCP.h': No such file or directory
> Errors were encountered while processing:
> /data/data/com.termux/files/usr/var/cache/apt/archives/ndk-sysroot_20-8_arm.deb
> E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)
> 
> $ termux-info
> Packages CPU architecture:
> arm
> Android version:
> 9
> Kernel build information:
> Linux localhost 3.4.113-g263ce0f96fc #1 SMP PREEMPT Sun Apr 21 19:18:10 CEST 2019 armv7l Android
> Device manufacturer:
> motorola
> Device model:
> XT1039

Please help me

**Comment by marcdw1289** [comment]
In the same boat here. Can't install ndk-sysroot. Was trying to install rust in which ndk is a prerequisite. Coincidentally, also a Motorola device as the last two users. Running OmniROM Oreo.

--- Errors ---

Preparing to unpack .../ndk-sysroot_20-8_aarch64.deb ...
Unpacking ndk-sysroot (20-8) ...
dpkg: error processing archive /data/data/com.termux/files/usr/var/cache/apt/archives/ndk-sysroot_20-8_aarch64.deb (--unpack):
 unable to open '/data/data/com.termux/files/usr/include/linux/netfilter/xt_DSCP.h.dpkg-new': No such file or directory
Errors were encountered while processing:
 /data/data/com.termux/files/usr/var/cache/apt/archives/ndk-sysroot_20-8_aarch64.deb
E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)

--- Errors ---

termux-info
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://termux.org/packages/ stable main
# science-repo (sources.list.d/science.list)
deb https://dl.bintray.com/grimler/science-packages-24 science stable
# game-repo (sources.list.d/game.list)
deb https://dl.bintray.com/grimler/game-packages-24 games stable
Updatable packages:
All packages up to date
Android version:
8.1.0
Kernel build information:
Linux localhost 3.18.116-Drunk_Optimus_Sanders #3 SMP PREEMPT Tue Jul 31 16:31:15 IST 2018 armv8l Android
Device manufacturer:
motorola
Device model:
XT1806

**Comment by marcdw1289** [comment]
Interesting finding. It seems that ndk-sysroot fails with certain ROMs.
Moto G5s potter arm, OmniROM Nougat. I multiboot and it installed fine with Tesla N.
Moto G5s Plus sanders arm64, OmniROM Oreo.

No issues on Axon 7 (AospExtended Nougat) or Essential PH-1 (/e/OS Pie).

Any reason why it would fail on OmniROM?

**Comment by bit-garden** [comment]
For me this issue started after updating the repos using the instructions here. https://wiki.termux.com/wiki/New_package_repository. Additionally I had to manually update the sources.list since it only succeeded in uninstalling everything.

This broke my chromebook and phones installs. I wiped and re-installed my chrombooks and is working fine now. My phone has a lot more particulars to set back up and would rather not go through the rebuild.

`apt purge ndk-sysroot` failed noting it wasn't installed.
`apt install ndk-sysroot` failed with 
```
trying to overwrite '/data/data/com.termux/files/usr/lib/libutil.so', which is also in package libutil 0.4-1
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 /data/data/com.termux/files/usr/var/cache/apt/archives/ndk-sysroot_20-8_arm.deb
```

**Comment by Grimler91** [comment]
@bit-garden if you have a android >= 7 device and you want to switch from the old repositories to the new then you have to uninstall all installed packages, change the source.list url and then install everything again. 
Reinstalling the app does the first 2 steps. We also have the `termux-upgrade-repo` script that does all three steps this while keeping your files in $HOME safe. What you cannot cannot do is to just edit sources.list and proceed as usual, you will then have a mix of packages without RPATH and packages with RPATH which will lead to problems.

Easiest fix for you now would be to ` rm -rf $PREFIX` and then restart the app. This will remove all installed packages and re-install the bootstrap packages but leave $HOME intact.

**Comment by bit-garden** [comment]
I'm on android 7.1.2. 

I did use the termux-upgrade-repo script which uninstalled and re-installed everything but from the old repo. 

I may have not been so clear in the last post. I changed the sources.list file and then did the termux-upgrade-repo script again which is where I'm at now. 

This issue kind of issue happens to me frequently on my Ubuntu PC, to where I can no longer install or even use apt-get because of some broken dependency. So I can't exactly say I didn't cause it ^_^

---

I'm in the process of backing up my home folder now and unless you have an idea to test(for information sake) I can reset from scratch. It seems my Chromebook instance has had no issues so far after wiping/re-installing.
